src/sage/combinat/integer_vectors_mod_permgroup.py

```diff
@@ -775,6 +785,79 @@ def __iter__(self):
             else:
                 return SF.elements_of_depth_iterator(self._sum)
 
+    def cardinality(self):
+        r"""Return the number of integer vectors in the set.
+
+        The number is computed using the cycle index theorem, which is
+        faster than listing the vectors.
+
+        EXAMPLES::
+
+            sage: G = PermutationGroup([], domain=[1,2,3])
+            sage: IntegerVectorsModPermutationGroup(G, 5).cardinality()
+            21
+
+            sage: G = PermutationGroup([(1,2)])
+            sage: IntegerVectorsModPermutationGroup(G, 1000).cardinality()
+            501
+
+            sage: G = PermutationGroup([(1,2,3)])
+            sage: I = IntegerVectorsModPermutationGroup(G, sum=10, max_part=5)
+            sage: I.cardinality()
+            7
+
+            sage: G = SymmetricGroup(10)
+            sage: I = IntegerVectorsModPermutationGroup(G, max_part=1)
+            sage: I.cardinality()
+            11
+        """
+        G = self._permgroup
+        k = G.degree()          # Vector length
+        d = self._sum           # Required total
+        m = self._max_part      # Max of one entry, -1 for no limit
+        if m == -1:
+            m = d               # Any entry cannot exceed total
+        if k == 0:
+            # Special case: Empty vectors cannot have nonzero sum.
+            if d == 0 or d is None:
+                return Integer(1)
+            else:
+                return Integer(0)
+
+        if d is None:
```

Comment:
Sorry, I made a silly mistake.  I should have written `sum(c*(m+1)^len(cycle_type) for cycle_type, c in G.cycle_index())`.  Is that right?

Reply:
Yes, that is correct. This is because the function-counting series is then just a `k*m`-degree polynomial, and the sum of its coefficients is the value of the polynomial at `x==1`. (There is probably a more direct combinatorial way to see this also.)  This adds a bit of complication/duplication to the code, but I guess it is worth it because it reduces to simple integer arithmetic, avoiding power series computations altogether in the `d is None` case.